### PR TITLE
log git branch

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
it was not logged before. This is fixed by this pr

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: yes
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: yes
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: yes
4. Have you successfully executed the linter and tests?
   - Answer: yes
5. Are all functions concise and don't require splitting?
   - Answer: yes
6. Is there documentation for arguments and complex logic?
   - Answer: yes
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer: yes
8. Are all functions placed in appropriate files?
   - Answer: yes
9. Is the PR name meaningful and descriptive?
   - Answer: yes
10. Is PR description provided, or unnecessary?
    - Answer: yes
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer:
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer:
13. Is there a need for a second reviewer?
    - Answer:

### CODE REVIEW

File: lizrd/support/code_versioning.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def version_code(self):
...
             self.original_branch_commit_hash = self.repo.head.object.hexsha
 
+            # check if the branch is tracking any remote branch at origin
+            assert (
+                self.repo.active_branch.tracking_branch() is not None
+            ), "Branch is not tracking any remote branch. Aborting..."
+            assert (
+                self.repo.active_branch.tracking_branch().remote_name == "origin"
+            ), "Branch is not tracking any remote branch. Aborting..."
+
```
Comment (crewtool) [inline lizrd/support/code_versioning.py:51]: Można jako jedna komenda.

File: lizrd/scripts/grid.py [hunk 2]
```diff
@@ -37,12 +37,21 @@
     except IndexError:
         raise ValueError("No config path specified. Aborting...")
 
+    try:
+        code_mirror_git_branch = sys.argv[2]
+    except IndexError:
+        code_mirror_git_branch = ""
+        assert (
+            CLUSTER_NAME == MachineBackend.LOCAL
+        ), f"No git branch with mirrored code specified. If you are calling grid.py directly on a cluster (not recommneded), supply any string as the second argument. Aborting..."
```
Comment (crewtool) [inline lizrd/scripts/grid.py:46]: 'recommended' spelling mistake

Comment (crewtool) [reply lizrd/scripts/grid.py:46]: Also, I don't like passing git_branch_name as a param here. How about passing it in yaml?

File: lizrd/scripts/run_exp_remotely.sh [hunk 3]
```diff
@@ -9,8 +9,10 @@ set -e
 source venv/bin/activate
 # run your python script
 python3 -m lizrd.support.sync_and_version --host $1
-base_dir=$(cat base_dir.txt)
-rm base_dir.txt
+base_dir=$(cat /tmp/base_dir.txt)
+git_branch=$(cat /tmp/git_branch.txt)
+rm /tmp/base_dir.txt
+rm /tmp/git_branch.txt
```
Comment (crewtool) [inline lizrd/scripts/run_exp_remotely.sh:15]: Co jeśli 2 procesy jednocześnie będą chciały skorzystyać z /tmp/base_dir.txt

File: lizrd/support/code_versioning.py [hunk 4]
```diff
@@ -60,15 +68,18 @@ def version_code(self):
...
             )
 
-        except GitCommandError:
+        except GitCommandError as e:
+            print(
+                f"Git command error encountered. Trying to revert changes... Git error message: {e}"
+            )
             self.handle_failure()
-            raise Exception("Failed to version code. Aborting...")
+            raise Exception("Changes successfully reverted. Aborting...")
```
Comment (crewtool) [inline lizrd/support/code_versioning.py:82]: Rzucanie wyjątku przy sukcesie?

File: lizrd/support/sync_and_version.py [hunk 5]
```diff
@@ -120,8 +121,14 @@ def set_up_permissions(host):
...
         f.write(base_dir)
     set_up_permissions(args.host)
-    name_for_branch = "exp_" + generate_random_string(10)
+    name_for_branch = (
+        "exp_"
+        + f"{datetime.datetime.now().strftime('%Y-%m-%d-%H-%M-%S')}"
+        + generate_random_string(10)
+    )
+    with open("/tmp/git_branch.txt", "w") as f:
+        f.write(name_for_branch)
```
Comment (crewtool) [inline lizrd/support/sync_and_version.py:133]: Poszukajmy jeszcze lepszego rozwiązania na to

**Review by crewtool — COMMENTED** [review]
do przegadania

File: lizrd/support/sync_and_version.py (hunk 5 above)
Comment (Simontwice) [reply lizrd/support/sync_and_version.py:133]: ok pogadamy

File: lizrd/scripts/grid.py (hunk 2 above)
Comment (Simontwice) [reply lizrd/scripts/grid.py:46]: pogadamy

File: lizrd/scripts/run_exp_remotely.sh (hunk 3 above)
Comment (Simontwice) [reply lizrd/scripts/run_exp_remotely.sh:15]: fair point

Comment (Simontwice) [reply lizrd/scripts/run_exp_remotely.sh:15]: i mean not really, to nie są procesy - to my puszczamy te rzeczy z palca. Nikt nie będzie puszczał lokalnie dwóch jobów na raz, bez przesady. Puszczanie joba trwa 30 sekund.

File: lizrd/scripts/grid.py [hunk 6]
```diff
@@ -37,12 +37,21 @@
     except IndexError:
         raise ValueError("No config path specified. Aborting...")
 
+    try:
+        code_mirror_git_branch = sys.argv[2]
```
Comment (maciek-pioro) [inline lizrd/scripts/grid.py:41]: Let's introduce argparse already xd

**Review by maciek-pioro — CHANGES_REQUESTED** [review]
One comment about argparse in grid

File: lizrd/scripts/grid.py (hunk 6 above)
Comment (Simontwice) [reply lizrd/scripts/grid.py:41]: you mean lizardparse

Comment (Simontwice) [reply lizrd/scripts/grid.py]: done

**Review by maciek-pioro — APPROVED** [review]

**Pull request merged**